### PR TITLE
ign -> gz : ros_gz Migration (Infra, Docs, and CMake)

### PR DESCRIPTION
**Pull request by methylDragon**
Part of: https://github.com/gazebosim/ros_gz/pull/262

Migrates:
- Docs and most comments
- GitHub actions
- IGNITION_VERSION env var
- CMake internal variables that are independent of project name

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,24 +25,24 @@ Rolling | Garden (not released) | [ros2](https://github.com/gazebosim/ros_gz/tre
 This repository holds packages that provide integration between
 [ROS](http://www.ros.org/) and [Gazebo](https://gazebosim.org):
 
-* [ros_ign](https://github.com/gazebosim/ros_gz/tree/ros2/ros_ign):
+* [ros_gz](https://github.com/gazebosim/ros_gz/tree/ros2/ros_gz):
```
Comment (chapulina) [inline README.md:28]: Just leaving a note that these links will be broken until the packages are migrated

File: ros_ign_bridge/src/convert/sensor_msgs.cpp [hunk 2]
```diff
@@ -413,7 +413,7 @@ convert_ign_to_ros(
   convert_ign_to_ros(ign_msg.header(), ros_msg.header);
   convert_ign_to_ros(ign_msg.field_tesla(), ros_msg.magnetic_field);
 
-  // magnetic_field_covariance is not supported in Ignition::Msgs::Magnetometer.
+  // magnetic_field_covariance is not supported in ignition::Msgs::Magnetometer.
```
Comment (chapulina) [inline ros_ign_bridge/src/convert/sensor_msgs.cpp]: 🤔

File: ros_ign_image/test/test_utils.h [hunk 3]
```diff
@@ -137,7 +137,7 @@ namespace testing
   }
 
   //////////////////////////////////////////////////
-  /// Ignition::msgs test utils
+  /// Gazebo::msgs test utils
```
Comment (chapulina) [inline ros_ign_image/test/test_utils.h]: Humm either

```suggestion
  /// Gazebo msgs test utils
```

or

```suggestion
  /// gz::msgs test utils
```

File: ros_ign_point_cloud/examples/depth_camera.sdf [hunk 4]
```diff
@@ -1,16 +1,17 @@
...
   1. Start ROS core:
 
           roscore
 
-  2. Make sure Ignition Gazebo can find the point cloud plugin by adding your lib install directory
+  2. Make sure Gazebo Sim can find the point cloud plugin by adding your lib install directory
      to the path, for example:
 
+          export GZ_SIM_SYSTEM_PLUGIN_PATH=<ws>/install/lib/
           export IGN_GAZEBO_SYSTEM_PLUGIN_PATH=<ws>/install/lib/
```
Comment (chapulina) [inline ros_ign_point_cloud/examples/depth_camera.sdf]: Just the new one should be enough, we shouldn't encourage users to use the old one anymore. The same goes for other files.

File: ros_ign_gazebo/CMakeLists.txt [hunk 5]
```diff
@@ -14,57 +14,63 @@ find_package(ament_cmake REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(std_msgs REQUIRED)
 
+# TODO(CH3): Deprecated. Remove on tock.
+if("$ENV{GZ_VERSION}" STREQUAL "" AND NOT "$ENV{IGNITION_VERSION}" STREQUAL "")
+  message(DEPRECATION "Environment variable [IGNITION_VERSION] is deprecated. Use [GZ_VERSION] instead.")
+  set(ENV{GZ_VERSION} ENV{IGNITION_VERSION})
```
Comment (chapulina) [inline ros_ign_gazebo/CMakeLists.txt]: This wasn't working for me, it works like this:

```suggestion
  set(ENV{GZ_VERSION} $ENV{IGNITION_VERSION})
```

File: ros_ign_bridge/package.xml [hunk 6]
```diff
@@ -24,16 +24,16 @@
   <depend>trajectory_msgs</depend>
 
   <!-- Garden -->
-  <depend condition="$IGNITION_VERSION == garden">gz-msgs9</depend>
-  <depend condition="$IGNITION_VERSION == garden">gz-transport12</depend>
+  <depend condition="$GZ_VERSION == garden or $IGNITION_VERSION == garden">gz-msgs9</depend>
```
Comment (chapulina) [inline ros_ign_bridge/package.xml:27]: Just a heads up that the syntax for these conditions can be finicky at times. It needs to be understood by `colcon`, `rosdep` and `bloom`. I've checked that `colcon` and `rosdep` work, but I think we'll only know for sure with `bloom` once we make a release

**Review by chapulina — CHANGES_REQUESTED** [review]

File: ros_ign_point_cloud/examples/depth_camera.sdf (hunk 4 above)
Comment (methylDragon) [reply ros_ign_point_cloud/examples/depth_camera.sdf]: Is this the case even if those env vars aren't supported by distributions pre-Garden? :o

File: ros_ign_bridge/src/convert/sensor_msgs.cpp (hunk 2 above)
Comment (methylDragon) [reply ros_ign_bridge/src/convert/sensor_msgs.cpp]: I'll migrate all ignition:: to gz:: even though we're not migrating the namespaces in source

File: ros_ign_point_cloud/examples/depth_camera.sdf (hunk 4 above)
Comment (methylDragon) [reply ros_ign_point_cloud/examples/depth_camera.sdf]: I'll remove IGN_GAZEBO_SYSTEM_PLUGIN_PATH anyway

Comment (chapulina) [reply ros_ign_point_cloud/examples/depth_camera.sdf]: Ahh good call, I forgot that this branch supports Fortress. We'll backport the env vars upstream soon 🤞🏽

**Review by chapulina — APPROVED** [review]

**Pull request merged**